dvc/repo/add.py

```diff
@@ -1,18 +1,35 @@
 import os
+import logging
+import colorama
 
 from dvc.repo.scm_context import scm_context
 from dvc.stage import Stage
-from dvc.utils import walk_files
+from dvc.utils import walk_files, LARGE_DIR_SIZE
 from dvc.exceptions import RecursiveAddingWhileUsingFilename
 
 
+logger = logging.getLogger(__name__)
+
+
 @scm_context
 def add(repo, target, recursive=False, no_commit=False, fname=None):
     if recursive and fname:
         raise RecursiveAddingWhileUsingFilename()
 
     targets = _find_all_targets(repo, target, recursive)
 
+    if os.path.isdir(target) and len(targets) > LARGE_DIR_SIZE:
+        logger.warning(
+            "You are adding a large directory recursively,"
```

Comment:
Should we tell which directory user is adding?

Reply:
good suggestion :+1: .